File: client-sdk/python/voice-dialogue-agent/sdk_usage.py
from __future__ import annotations

from argparse import Namespace

from eva_client_sdk import (
    AsrConfig,
    BargeInConfig,
    CameraConfig,
    EmotionConfig,
    EvaVoiceDialogueAgentConfig,
    LlmConfig,
    MediaTransports,
    TtsConfig,
    VadConfig,
    create_eva_voice_dialogue_agent,
)
from eva_client_sdk.contracts import StaticGreeting
from eva_client_sdk.media import OpenCvCameraSource, create_pyaudio_media_transports

from audio_setup import AudioSelection, MeteredInputSource, create_demo_aec
from commands import build_demo_commands
from event_output import show_event
from settings import CAMERA_SYSTEM_PROMPT, GREETING_PROFILES


async def start_eva_agent(*, api_key: str, args: Namespace, audio: AudioSelection):
    """集中展示 Python 语音 demo 的 SDK 接入主路径。"""

    aec = create_demo_aec(
        audio.input_sample_rate,
        tts_sample_rate=args.tts_sample_rate,
    )
    pyaudio = create_pyaudio_media_transports(
        input_sample_rate=audio.input_sample_rate,
        input_device_index=audio.input.index,
        output_device_index=audio.output.index,
        aec=aec,
    )
    transports = MediaTransports(
        input=MeteredInputSource(pyaudio.input),
        output=pyaudio.output,
        aec=pyaudio.aec,
        camera=(
            OpenCvCameraSource(jpeg_quality=70, max_long_edge=640)
            if args.camera
            else None
        ),
    )

    # SDK 接入核心：model、VAD、barge-in、command、Emotion、camera 与 Media SPI。
    agent = create_eva_voice_dialogue_agent(
        EvaVoiceDialogueAgentConfig(
            api_key=api_key,
            asr=AsrConfig(model="ark-asr-plus", sample_rate=16_000),
            llm=LlmConfig(
                model="volcengine-doubao-seed-2.0-mini",
                extra_parameters={"thinking": {"type": "disabled"}},
            ),
            tts=TtsConfig(
                model=args.tts_model,
                voice=args.tts_voice,
                sample_rate=args.tts_sample_rate,
            ),
            vad=VadConfig(sensitivity=0.7, silence_threshold_ms=400),
            barge_in=BargeInConfig(
                initial_playback_guard_ms=args.initial_playback_guard_ms
            ),
            greeting=StaticGreeting(text=GREETING_PROFILES[args.greeting_profile]),
            commands=build_demo_commands(),
            emotion=EmotionConfig(enabled=not args.no_emotion),
            camera=CameraConfig() if args.camera else None,
            system_prompt=CAMERA_SYSTEM_PROMPT,
            transports=transports,
        )
    )

    unsubscribe = agent.on_event(show_event)
    try:
        if args.camera:
            await agent.set_camera_enabled(True)
            print("camera: enabled", flush=True)
        await agent.start()
        return agent, unsubscribe
    except BaseException:
        unsubscribe()
        await agent.stop()
        raise

File: client-sdk/cpp/voice-dialogue-agent/src/sdk_usage.cpp
#include "sdk_usage.hpp"
#include "commands.hpp"
#include <eva/helpers.hpp>

eva::Result<std::unique_ptr<eva::Agent>> create_agent(std::string key, const Options& options) {
  using namespace eva;
#if defined(__APPLE__)
  auto input = make_portaudio_audio_input({});
  auto output = make_portaudio_audio_output(PortAudioOutputOptions{std::nullopt, options.output_gain});
#else
  auto input = make_alsa_audio_input(AlsaInputOptions{options.input_device});
  auto output = make_alsa_audio_output(AlsaOutputOptions{options.output_device, options.output_gain});
#endif
  if (!input) return Result<std::unique_ptr<Agent>>::failure(input.error());
  if (!output) return Result<std::unique_ptr<Agent>>::failure(output.error());
  auto aec = make_native_aec(NativeAecOptions{options.aec_delay_ms});
  if (!aec) return Result<std::unique_ptr<Agent>>::failure(aec.error());
  std::unique_ptr<CameraSnapshotSource> camera;
  if (options.camera) {
#if defined(__APPLE__)
    auto made = make_avfoundation_camera(AvFoundationCameraOptions{options.camera_device, 640, 70});
#else
    auto made = make_v4l2_camera(V4l2CameraOptions{options.camera_device});
#endif
    if (!made) return Result<std::unique_ptr<Agent>>::failure(made.error());
    camera = std::move(made).value();
  }
  AgentConfig config;
  config.api_key = std::move(key);
  config.asr = {"ark-asr-plus", 16000};
  config.llm.model = "volcengine-doubao-seed-2.0-mini";
  config.llm.extra_parameters = {{"thinking", JsonValue::object({
      {"type", JsonValue::string("disabled").value()}}).value()}};
  config.tts.model = "ark-tts-flash";
  config.tts.voice = "zh_female_kido";
  config.tts.sample_rate = 44100;
  config.vad = VadConfig{0.7, 400};
  config.barge_in = BargeInConfig{options.initial_playback_guard_ms};
  config.history = HistoryConfig{10};
  config.greeting = StaticGreeting{"你好，我是 EVA。很高兴见到你，你可以告诉我现在想聊什么。"};
  config.system_prompt = "你是 EVA。请用自然中文简短回答，最多两句话，不使用 Markdown。";
  config.emotion = EmotionConfig{true};
  CommandsConfig commands;
  commands.registrations.push_back(make_clock_command());
  commands.registrations.push_back(make_indicator_command(std::make_shared<IndicatorState>()));
  config.commands = std::move(commands);
  config.transports.emplace(MediaTransports{std::move(input).value(), std::move(output).value(),
      std::move(aec).value(), std::move(camera)});
  if (options.camera) config.camera = CameraConfig{};
  return Agent::create(std::move(config));
}
